cmd/kanikoExecute.go

```diff
@@ -144,3 +149,10 @@ func certificateUpdate(certLinks []string, httpClient piperhttp.Sender, fileUtil
 	}
 	return nil
 }
+
+func generateDockerConfigJSON(username, password, registryURL string) []byte {
+	usernamePassword := fmt.Sprintf("%s:%s", username, password)
+	encodedUsernamePassword := base64.StdEncoding.EncodeToString([]byte(usernamePassword))
+	configJSON := fmt.Sprintf(`{"auths":{"%s":{"auth":"%s"}}}`, registryURL, encodedUsernamePassword)
+	return []byte(configJSON)
+}
```

Comment:
This is already available in the internal repository, please have a look and extract package to reuse.

Reply:
Please re-review and gladly add your thoughts to the conversation. 

I haven't added the usage of the CreateDockerConfigJSON method to all the above steps. My fear is the PR will grow and not be so easily reviewable. Instead I am willing to split the PR into its parts: One PR only with the immigration of the `CreateDockerConfigJSON` method and then the one PR for each step using it... What do you think?